### PR TITLE
Removed tests out of graph mode in weight decay optimizer test

### PR DESCRIPTION
**Pull request by autoih**
cc @gabrieldemarmiesse

### CODE REVIEW

File: tensorflow_addons/optimizers/tests/weight_decay_optimizers_test.py [hunk 1]
```diff
@@ -23,165 +24,125 @@
 WEIGHT_DECAY = 0.01
 
 
-class OptimizerTestBase(tf.test.TestCase):
-    """Base class for optimizer tests.
+# Optimizer tests may inherit from this class and define test functions using doTest. Usually this should include the functions testSparse, testBasic, and testBasicCallableParams. See weight_decay_optimizers_test for an example.
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/optimizers/tests/weight_decay_optimizers_test.py:27]: This comment can be removed now.

File: tensorflow_addons/optimizers/tests/weight_decay_optimizers_test.py [hunk 2]
```diff
@@ -212,129 +173,159 @@ def sgdw_update_numpy(param, grad_t, slot_vars, learning_rate, momentum, weight_
     return param_t, slot_vars
 
 
-@test_utils.run_all_in_graph_and_eager_modes
-class AdamWTest(OptimizerTestBase):
+optimizer_AdamW = weight_decay_optimizers.AdamW
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/optimizers/tests/weight_decay_optimizers_test.py:176]: We don't need an alias there, it actually hurts the readability. Please use `weight_decay_optimizers.AdamW` directly.

File: tensorflow_addons/optimizers/tests/weight_decay_optimizers_test.py [hunk 3]
```diff
@@ -212,129 +173,159 @@ def sgdw_update_numpy(param, grad_t, slot_vars, learning_rate, momentum, weight_
...
-        self.doTest(
-            self.optimizer,
-            sgdw_update_numpy,
-            do_decay_var_list=True,
-            learning_rate=0.001,
-            momentum=0.9,
-            weight_decay=WEIGHT_DECAY,
-        )
 
+optimizer_SGDW = weight_decay_optimizers.SGDW
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/optimizers/tests/weight_decay_optimizers_test.py:265]: Same here please use it directly.

File: tensorflow_addons/optimizers/tests/weight_decay_optimizers_test.py [hunk 4]
```diff
@@ -212,129 +173,159 @@ def sgdw_update_numpy(param, grad_t, slot_vars, learning_rate, momentum, weight_
...
+    do_test(
+        dtype,
+        optimizer_SGDW,
+        sgdw_update_numpy,
+        learning_rate=lambda: 0.001,
+        momentum=lambda: 0.9,
+        weight_decay=lambda: WEIGHT_DECAY,
+    )
 
-class ExtendWithWeightDecayTest(SGDWTest):
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/optimizers/tests/weight_decay_optimizers_test.py:335]: The equivalent of this would be to run the SGDWTest tests with the decorator:

```python
@pytest.mark.parametrize(
    "optimizer", 
    [weight_decay_optimizers.SGDW, 
     weight_decay_optimizers.extend_with_decoupled_weight_decay(tf.keras.optimizers.SGD)])
```

Comment (autoih) [reply tensorflow_addons/optimizers/tests/weight_decay_optimizers_test.py:335]: Thanks @gabrieldemarmiesse. Updated, needs your review again.

**Comment by gabrieldemarmiesse** [comment]
Looks great, thanks!
